scripts/changelog/README.md

```diff
@@ -0,0 +1,68 @@
+# Auto-generate changelog for ksqlDB releases
+
+The ksqlDB repo uses conventional commits in order to assist with auto-generation of changelogs.
+
+## One-time setup
+
+Install the [tool](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) we use to for generating changelogs:
+```
+npm install -g conventional-changelog-cli
+```
+
+## Generate changelog
+
+Assuming you've completed the one-time setup above, steps for generating a changelog are as follows.
+
+Suppose you're interested in generating a changelog of all changes in ksqlDB 0.8.0 that are new as of ksqlDB 0.7.0. Refer to these versions as `NEW_VERSION` and `OLD_VERSION`, respectively. The same process applies for other new/old ksqlDB version combinations as well (e.g., 0.8.1 and 0.8.0, or 0.8.0 and 0.7.1).
+
+These steps assume you're in the root of the `ksql` directory.
+
+### List commits on `NEW_VERSION`
+
+Start by listing all commits from the `NEW_VERSION`, which requires checking out the relevant branch/tag/commit. If the release tag exists:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.8.0-ksqldb -b v0.8.0-ksqldb
+```
+If the tag hasn't been created yet but a branch has been cut (e.g., if a release is in progress):
+```
+git checkout 0.8.x-ksqldb
+git fetch upstream
+git merge upstream/0.8.x-ksqldb
+```
+If the branch hasn't been cut yet (which could also be the case for a release currently in progress):
+```
+git checkout <GIT_SHA>
+```
+
+Once on the relevant branch/tag/commit, fetch the commits and save them to a file:
+```
+conventional-changelog -p angular -i new_commits.txt -s -r 0
+```
+
+### List commits on `OLD_VERSION`
+
+Next, repeat the process to list commits from the `OLD_VERSION`. A tag for this release should already exist, so check out the tag:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.7.0-ksqldb -b v0.7.0-ksqldb
```

Comment:
This causes ambiguous errors in some environments when running the conventional-changelog. To make it work, I did not create the branch:
```
# git checkout tags/v0.9.0-rc294-ksqldb -b v0.9.0-rc294-ksqldb
Switched to a new branch 'v0.9.0-rc294-ksqldb'
[root@ksql-centos ksql]# conventional-changelog -p angular -i new_commits.txt -s -r 0
Error: Error in git-raw-commits: warning: refname 'v0.9.0-rc294-ksqldb' is ambiguous.
```
```
$ git checkout tags/v0.7.0-ksqldb
Note: switching to 'tags/v0.9.0-rc294-ksqldb'.
...
$ conventional-changelog -p angular -i new_commits.txt -s -r 0
(base) sergio@pop-os:~/Development/ksql$
```

Reply:
I'm still confused where this error is coming from and why it only affects your setup but not mine, but the workaround sounds good so I've updated the doc. Thanks!